compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -794,6 +797,21 @@ export class Environment {
               binding.imported,
             );
             if (importedType != null) {
+              /*
+               * Check that hook-like export names are hook types, and non-hook names are non-hook types.
+               * The user-assigned alias isn't decidable by the type provider, so we ignore that for the check.
```

Comment:
Maybe we can enforce that the isHookName of the exported name and the local alias are the same?

Reply:
I think that makes sense, but we should do that everywhere regardless of whether we have a type provider for the module or not. Let's explore that in a follow-up.